app/api/api_v1/endpoints/article.py

```diff
@@ -49,11 +62,56 @@ def create_article(
     *,
     db: Session = Depends(deps.get_db),
     article_in: schemas.ArticleCreate,
-    current_user=Depends(deps.get_current_active_admin_user),
+    current_user=Depends(deps.get_current_active_user),
 ) -> Any:
     """
     Create new user.
     """
+    if current_user.user_role != UserRoleEnum.ADMIN or current_user.user_role != UserRoleEnum.VOLUNTEER:
+        raise HTTPException(status_code=403, detail="Not enough permissions")
+    article_category = crud.article_category.get(db, id=article_in.article_category_id)
+    if not article_category:
+        raise HTTPException(status_code=404, detail="Article category not found")
+    
     article = crud.article.create(db, obj_in=article_in, user=current_user)
 
     return article
+
+@router.put("/{article_id}", response_model=schemas.Article)
+def update_article(
+    *,
+    db: Session = Depends(deps.get_db),
+    article_id: int,
+    article_in: schemas.ArticleUpdate,
+    current_user=Depends(deps.get_current_active_user),
+) -> Any:
+    """
+    Update article.
+    """
+    article = crud.article.get(db, id=article_id)
+    if not article:
+        raise HTTPException(status_code=404, detail="Article not found")
+    article_category = crud.article_category.get(db, id=article_in.article_category_id)
```

Comment:
tutaj też musimy się upewnić, że użytkownik może edytować tylko swój artykuł (lub jeśli jest adminem to może wszystkie), wydaje mi się, że wystarczy dodać to samo co przy usuwaniu:
```suggestion
    article_category = crud.article_category.get(db, id=article_in.article_category_id)
    if article.created_by_id != current_user.id or current_user.user_role != UserRoleEnum.ADMIN:
        raise HTTPException(status_code=403, detail="Not enough permissions")
```

Reply:
Dodane